test-rpc-functions/src/rpc.rs

```diff
@@ -269,6 +280,59 @@ impl RpcTestFunctionsRpcServer for super::RpcTestFunctionsHandle {
 
         Ok(vrf_output.hex_encode())
     }
+
+    async fn generate_transactions(
+        &self,
+        mut input_tx_id: Id<Transaction>,
+        num_transactions: u32,
+        amount_to_spend: u64,
+        fee_per_tx: u64,
+    ) -> rpc::Result<Vec<HexEncoded<SignedTransaction>>> {
+        let coin_decimals = self
+            .call(|this| this.get_chain_config().map(|chain| chain.coin_decimals()))
+            .await
+            .expect("Subsystem call ok")
+            .expect("Subsystem call ok");
+        let coin_decimal_factor = pow10(coin_decimals);
+        let mut amount_to_spend = (amount_to_spend as u128) * coin_decimal_factor;
+        let fee_per_tx = (fee_per_tx as u128) * coin_decimal_factor;
+        let mut transactions = vec![];
+        for _ in 0..num_transactions {
+            let inputs = vec![TxInput::from_utxo(OutPointSourceId::Transaction(input_tx_id), 0)];
+            let mut outputs = vec![TxOutput::Transfer(
+                OutputValue::Coin(Amount::from_atoms(amount_to_spend)),
+                Destination::AnyoneCanSpend,
+            )];
+            outputs.extend((0..9999).map(|_| {
+                TxOutput::Transfer(
+                    OutputValue::Coin(Amount::from_atoms(1)),
+                    Destination::AnyoneCanSpend,
+                )
+            }));
+
+            let transaction = SignedTransaction::new(
+                Transaction::new(0, inputs, outputs).expect("should not fail"),
+                vec![InputWitness::NoSignature(None)],
+            )
+            .expect("num signatures ok");
+
+            input_tx_id = transaction.transaction().get_id();
+            amount_to_spend -= 10000;
+            amount_to_spend -= fee_per_tx;
+
+            transactions.push(HexEncoded::new(transaction));
+        }
+
+        Ok(transactions)
+    }
+}
+
+fn pow10(coin_decimals: u8) -> u128 {
+    let mut coin_decimal_factor = 1;
```

Comment:
Why not just `10u128.pow(coin_decimals)`?

Reply:
fixed